src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs

```diff
@@ -230,7 +236,10 @@ private bool GetDocuments(IServiceProvider services)
             return false;
         }
 
-        var documentNames = (IEnumerable<string>)InvokeMethod(getDocumentsMethod, service, GetDocumentsArguments);
+        // If an explicit document name is provided, then generate only that document.
+        var documentNames = _context.DocumentName != null
```

Comment:
Would it be worthwhile to retrieve the documents list regardless and then filter to the specified name (i.e. to confirm that it's valid)?

Reply:
Good call -- can do.